kernel/src/net/socket/ip/stream/mod.rs

```diff
@@ -657,18 +663,32 @@ impl Socket for StreamSocket {
                 let no_delay = options.tcp.no_delay();
                 tcp_no_delay.set(no_delay);
             },
-            tcp_congestion: Congestion => {
-                let congestion = options.tcp.congestion();
-                tcp_congestion.set(congestion);
-            },
             tcp_maxseg: MaxSegment => {
                 let maxseg = options.tcp.maxseg();
                 tcp_maxseg.set(maxseg);
             },
+            tcp_keep_idle: KeepIdle => {
+                let keep_idle = options.tcp.keep_idle();
+
+                let seconds = {
+                    let now = Jiffies::elapsed().as_duration();
+                    if let Some(duration) = keep_idle.and_then(|keep_idle| keep_idle.checked_sub(now)) {
```

Comment:
Why do this? It does not seem right to me.

Getting `TCP_KEEPIDLE` should return the original value that was set before, right?